app/components/chat_component.html.erb

```diff
@@ -0,0 +1,14 @@
+<div class='active-admin-chat__conversation-history-container'>
+  <div class='active-admin-chat__conversation-history' data-motion='scroll->load_messages'>
+    <% if conversation && messages.empty? %>
+      <div class='active-admin-chat__conversation-history no-messages'>No messages</div>
+    <% else %>
+      <%= render partial: 'active_admin/chat/messages', locals: { messages: messages } %>
```

Comment:
Future optimization, render only new messages

Reply:
This will probably not ever be possible with Motion alone.

To be fair, I also don't think this is possible in React with function components.

Reply:
A teammate told me that Phoenix LiveView solved it like this https://hexdocs.pm/phoenix_live_view/dom-patching.html , as far as I understand it's a different mode when there is a lot of data where you send from the server only data to be appended or prepended